gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql

```diff
@@ -0,0 +1,124 @@
+/* gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql */
+
+/*
+ This is part of arenadata_toolkit API for ADB Bundle.
+ This function collects information in db_files_current and db_files_history tables.
+ */
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_collect_table_stats()
+RETURNS VOID
+AS $$
+BEGIN
+	IF (
+		SELECT NOT bool_or(val) FROM (

```

Comment:
Horror! How about a request like this?
```sql
if not exists (
    select
    from pr_filtered
    where now() between ...
)
```

Reply:
No, with query like
```
	IF NOT EXISTS (
			SELECT
			FROM ( 
				SELECT pr.parrangestart, pr.parrangeend, pr.parchildrelid 
				FROM pg_partition pp 
				JOIN pg_class cl1 ON pp.parrelid = cl1.oid 
				JOIN pg_partition_rule pr ON pr.paroid = pp.oid 
				JOIN pg_class cl2 ON cl2.oid = pr.parchildrelid 
				JOIN pg_namespace n1 ON cl1.relnamespace = n1.oid 
				JOIN pg_namespace n2 ON cl2.relnamespace = n2.oid 
				WHERE 
					pp.paristemplate = false 
					AND 
					n1.nspname = 'arenadata_toolkit' 
					AND 
					n2.nspname = 'arenadata_toolkit' 
					AND 
					cl1.relname = 'db_files_history' 
					AND 
					pp.parkind = 'r'  --range 
					AND 
					pr.parrangestartincl = 't' ) pr_filtered
			WHERE
				now() BETWEEN 
				CAST(substring(pg_get_expr(pr_filtered.parrangestart, pr_filtered.parchildrelid) FROM '#"%#"::%' FOR '#') 
					AS TIMESTAMP WITHOUT TIME ZONE) 
				AND 
				CAST(substring(pg_get_expr(pr_filtered.parrangeend, pr_filtered.parchildrelid) FROM '#"%#"::%' FOR '#') 
					AS TIMESTAMP WITHOUT TIME ZONE))
	THEN
```
is planned in a way that the filter for partition range is applied first. And we get the error.

Reply:
```sql
if not exists (
    with pr_filtered as (
        ...
    ) select from pr_filtered where now() between ...
)
```

Reply:
And making 'pr_filtered' CTE also doesn't make any difference.

Reply:
then consider using opclass
```sql
select opcname from pg_opclass where oid in (select parclass[0] from pg_partition)
```

Reply:
will try

Reply:
It looks that using 'case' in the 'where' filter also helps.